### ISSUE TITLE
Microsite generator

### ISSUE THREAD

**Issue by vyedin** [techforwarren/start-here#3]
Often in the day-to-day of the campaign, there's a big news moment that spurs a lot of interest, and it's worthwhile to spin up a quick microsite to capture enthusiasm generated by the moment. Generally the need in these sites is very simple (1-2 pages of a little content/video/imagery; signup form; donate button). It'd be nice to automate away as much of the drudgery as necessary - even up to and including domain name registration - so that someone in say the digital organizing operation could put one of these together in an hour.

**Comment by john-osullivan** [comment]
What stack is the campaign on?  Would a Terraform solution standing on top of AWS get the job done, or would it be better to go lower-infra and use a platform like Zeit Now?

**Comment by john-osullivan** [comment]
Or taking a different approach, how code-savvy are the people in the digital organizing operation?  Would it be best to hook something up to a CMS like Wordpress, or do people want the flexibility of actually writing the code?  

If they want code, then we could set up a CLI which creates the boilerplate, infra, and domain with something like `create-warren-app issue-of-the-day this-issue-matters.com`.  That way they can focus on getting the page correct and everything else just works.

If they'd prefer no code, then we could set write automation which makes it easy to spin up a new web page for a subset of posts from a CMS.

**Comment by shaisachs** [comment]
Part of me thinks that a good way to start is not to create a generator, but rather to create a simple microsite for a specific event. So for example, build a really simple microsite focused on the September debates. But design it in such a way that it could easily be "templatized" and reused for other events. This approach is sort of the lean startup approach, if you will - start with the simplest possible solution, learn what you can about success in that realm, then pivot on to the next thing.

Along those lines I'm thinking a really simple Github Pages microsite. If the first one goes well then the next one can be a really simple copy/pasta operation, and if that succeeds we can start thinking about doing something more fancy for the next iteration.

Make sense?

**Comment by john-osullivan** [comment]
I can work with that!  Like the idea of making something that looks good and has all the core content cut out to a template.  My first guess would be that we could have special files for all the key content, so creating a new one within same template is just a matter of updating some Markdown text.

Do you wanna make the repo within here so everyone can see it and potentially help out?  Happy to just make it on my own GitHub if you'd rather not for now

**Comment by shaisachs** [comment]
Nice! Give https://github.com/techforwarren/septdebates a try, does that work for you?

**Comment by john-osullivan** [comment]
Perfect!  I've been doing some research on how to make the "including registering a domain name" part easy.  It looks like the easiest option would be Google Domains.

They've got this [neat little plugin](https://developers.google.com/domains/widget/) so the digital organizers could buy configured domains & create pages within the same UI.  Organizations have to apply to use it, they have a [signup form](https://docs.google.com/forms/d/e/1FAIpQLScQ4jU5xjvbc0qa9vj0ixNDeHwRkH9znW8-CbeA2Jk_Barx3Q/viewform?c=0&w=1) .  They ask questions about users -- I think users here would mean the organizers, rather than the people using the sites.  This would be for the generator tool down the road, rather than this one for September.  We could also use DNSimple or Dynadot for this, they all seemed to have solid APIs 👍 

None of that is pressing, I'll focus on getting a sample page ready with the right pieces & styling for September

**Comment by john-osullivan** [comment]
Actually, I tried to push up late last night and got an authorization error -- could you please give me push access as well on these repos, @shaisachs ?  Thank you!

**Comment by shaisachs** [comment]
Hi there - I think the invite to `techforwarren` which I just sent you should resolve access.

**Comment by john-osullivan** [comment]
Hey -- took me a few days to sit down and put it together, but I've gathered most of the source material for this microsite!  The [repo's README](https://github.com/techforwarren/septdebates) has been updated with a rough prospective sketch and the content I'll be plugging in.  Going to try and get a first draft of the page out in the next couple days, shooting to have this ready to share by the end of the week!

**Comment by joegoldbeck** [comment]
@john-osullivan are you or anyone else charging ahead with this right now, or is this project somewhat up for grabs at the moment?

I'd love to see us get momentum on this and a team around it, with the goal of getting an MVP out the door in the near term, especially since this is a direct ask from the campaign :D

**Comment by danielnarey** [comment]
@john-osullivan Can you give us an update on what you’d like to do with the **septdebates** repo since the debate has happened already? Is this still something you’re working on, and should we rename the repo? 

@joegoldbeck @shaisachs

**Comment by shaisachs** [comment]
I should mention, this was sort of my idea- to put together a sample
microsite that we could templatize and then parlay into a generator. I
don't think it's a bad idea altogether, and we could certainly refocus on
October debates. But certainly happy to hear other approaches/ideas.

**Comment by danielnarey** [comment]
@shaisachs Yes, I got that from the thread. It might make sense to split this into two projects since @chrisabrams has told us that he has a prototype for a generator that is almost ready for public consumption. Let’s see if we can work with @john-osullivan on redefining the goal of the repository that he started - for example, developing the content and UI for now instead of worrying about implementing advanced tooling. 

Content-wise, we can shift to the next debate if that’s what people are motivated to work on, or we can hone in on another micro-topic that is relevant to the moment. I’m happy to give some content/design input as we move forward.
